File: java/575.java
class Solution {
    public int distributeCandies(int[] candies) {
        int l = candies.length;
        Set<Integer>candy = new HashSet();
        for(int i:candies){
            candy.add(i);
        }
        return Math.min(candy.size(),l/2);
    }
}

File: go/575.go
func distributeCandies(candies []int) int {
	ret := len(candies) / 2
	candy := make(map[int]int)
	for _, c := range candies {
		candy[c] = candy[c] + 1
	}
	if len(candy) < ret {
		return len(candy)
	}
	return ret
}